flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/StateValidator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.declarative;
+
+import org.apache.flink.util.Preconditions;
+
+import java.util.function.Consumer;
+
+/**
+ * Utility for state test classes (e.g. {@link WaitingForResourcesTest}) to track if correct input
+ * has been presented and if the state transition happened.
+ *
+ * @param <T> Type of the state to validate.
+ */
+public class StateValidator<T> {
+
+    private Runnable trap = () -> {};
+    private Consumer<T> consumer = null;
+    private final String stateName;
+
+    public StateValidator(String stateName) {
```

Comment:
have you thought about using the `State` class here?

Reply:
Nope, it's a nice idea (in case we rename states, etc.), however, this doesn't work right now, as this PR doesn't contain all required state classes yet (Executing, Restarting, ...). I can fix this with the last State PR ;)

Reply:
ah of course, nevermind then. We didn't want to couple the individual state tests to other states.